test/unit/parser/format_parse_test.cpp

```diff
@@ -1073,3 +1084,47 @@ TEST(parse_test, container_default)
         EXPECT_TRUE(option_values == (std::vector<int>{2, 1, 3}));
     }
 }
+
+TEST(parse_test, executable_name)
```

Comment:
Could you please add a test with a subparser if possible?

Reply:
There is a test with a sub parser, the 4th test is exactly that.